server-side/controllers/user.controller.js

```diff
@@ -1,34 +1,33 @@
-import mongoose  from 'mongoose';
+import mongoose from 'mongoose';
 import bcrypt from 'bcrypt';
 import Users from '../models/user.model.js';
 import {
   getUserById_service,
 } from '../services/user.service.js';
 
-export const getUsers = async (req, res,next) => {
+export const getUsers = async (req, res, next) => {
   try {
-    const users = await Users.find().populate('visitsWebsites profiles preference' ).select('-__v')
-    .select('-__v')
+    const users = await Users.find().populate({ path: 'visitsWebsites', populate: { path: 'websiteId' } }).populate('profiles preferences').select('-__v');
     res.status(200).send(users);
   } catch (err) {
     console.error(err);
-    next({message:err.message,status:500})
+    next({ message: err.message, status: 500 })
   }
 };
 
-export const getUserById = async (req, res,next) => {
-  const {id} = req.params;
-  if(!mongoose.Types.ObjectId.isValid(id))
-    return next({message:'id is not valid'})
+export const getUserById = async (req, res, next) => {
+  const { id } = req.params;
+  if (!mongoose.Types.ObjectId.isValid(id))
+    return next({ message: 'id is not valid' })
   try {
-    const user = await getUserById_service(id);    
+    const user = await Users.findById(id).populate('visitsWebsites profiles preferences').select('-__v');
```

Comment:
access to collections and logics should be in service file. why did you remove it?

Reply:
I changed it.